gammapy/cube/core.py

```diff
@@ -307,21 +307,26 @@ def wcs_pixel_to_skycoord(self, x, y, z):
         energy = self.energy_axis.wcs_pix2world(z)
         return (position, energy)
 
-    def to_sherpa_data3d(self, dstype='Data3D'):
+    def to_sherpa_data3d(self, dstype='Data3D', select_region=False, index_selected_region=None):
```

Comment:
I don't see why it's necessary to make the selection of the region here. Because it is done in `CombinedModel3DInt.calc()` again. Is it possible to remove the selection from `to_sherpa_data3d()`?